### PR TITLE
fix(test): handle NULL _irowts_origin in interp fill(near) test

### PR DESCRIPTION
**Pull request by facetosea**
- Add NULL check for _irowts_origin in check_result_for_near, validating _irowts, _isfilled and fill values when surroundingTime is exceeded
- Pass fill_values through query/check pipeline for proper verification
- Add 4 stable test cases for WHERE + range(point, duration) scenarios covering near/prev/next with NULL and non-NULL _irowts_origin

# Description

# Issue(s)

- Close/close/Fix/fix/Resolve/resolve: Issue Link

# Checklist

Please check the items in the checklist if applicable.

- [ ] Is the user manual updated?
- [ ] Are the test cases passed and automated?
- [ ] Is there no significant decrease in test coverage?

### CODE REVIEW

File: test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py [hunk 1]
```diff
@@ -6111,10 +6111,25 @@ def is_nearest(self, select_results, irowts_origin, irowts):
 
     ## interp_results: _irowts_origin, _irowts, ..., _isfilled
     ## select_all_results must be sorted by ts in ascending order
-    def check_result_for_near(self, interp_results, select_all_results, sql, sql_select_all):
+    ## fill_values: tuple of fill values, e.g. (1, 2), used to verify when _irowts_origin is NULL
+    def check_result_for_near(self, interp_results, select_all_results, sql, sql_select_all, fill_values=None):
         #tdLog.info(f"check_result_for_near for sql: {sql}, sql_select_all:{sql_select_all}")
         for row in interp_results:
             #tdLog.info(f"row: {row}")
+            if row[0] is None:
```
Comment (dapan1121) [inline test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py:6119]: 感觉这种校验方式不具备确定性

File: test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py [hunk 2]
```diff
@@ -6386,6 +6407,47 @@ def check_interp_fill_extension(self):
...
+        tdSql.checkData(0, 3, 12)
+        tdSql.checkData(0, 4, True)
+
+        ## near: WHERE narrows data to [10:00, 10:23], point at 09:30 (<1h from nearest data) → NOT NULL
+        sql = f"select _irowts_origin, _irowts, interp(c1), interp(c2), _isfilled from test.meters where ts between '2018-09-17 10:00:00' and '2018-09-17 10:23:20' range('2018-09-17 09:30:00', 1h) fill(near, 7, 8)"
+        tdSql.query(sql, queryTimes=1)
+        tdSql.checkRows(1)
+        if tdSql.queryResult[0][0] is None:
+            tdLog.exit(f"_irowts_origin should not be None when nearest data is within surroundingTime")
+        tdSql.checkData(0, 4, True)
```
Comment (dapan1121) [inline test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py:6482]: **[BLOCKER] 第4个用例断言严重不足，属于"跑出什么结果就是什么结果"**

这里只检查了 `_irowts_origin is not None` 和 `_isfilled=True`，但对这条测试路径（`_irowts_origin` 非 NULL）最关键的数据正确性完全没有验证：

| 列 | 字段 | 应有断言 | 当前状态 |
|---|---|---|---|
| col 0 | `_irowts_origin` | 等于最近真实数据点的精确 ts | 只判断 is not None |
| col 1 | `_irowts` | 等于插值点 `'2018-09-17 09:30:00.000'` | **完全未检查** |
| col 2 | `interp(c1)` | 等于最近真实数据点的实际 c1 值（不是 fill 默认值 7） | **完全未检查** |
| col 3 | `interp(c2)` | 等于最近真实数据点的实际 c2 值（不是 fill 默认值 8） | **完全未检查** |

对比前三个 NULL 用例，每列都有明确断言；这个用例只验证了 2 个字段，即使 interp 函数错误地返回任意值也会通过。

建议修改：先动态查出 WHERE 范围内最近数据点的精确值，再逐列断言：
```python
nearest_sql = ("select ts, c1, c2 from test.meters "
               "where ts between '2018-09-17 10:00:00' and '2018-09-17 10:23:20' "
               "order by ts asc limit 1")
tdSql.query(nearest_sql)
nearest_ts = tdSql.queryResult[0][0]
nearest_c1 = tdSql.queryResult[0][1]
nearest_c2 = tdSql.queryResult[0][2]

tdSql.checkData(0, 0, nearest_ts)                # _irowts_origin = 真实最近点 ts
tdSql.checkData(0, 1, '2018-09-17 09:30:00.000') # _irowts = 插值点本身
tdSql.checkData(0, 2, nearest_c1)                # interp(c1) = 真实值，非 fill=7
tdSql.checkData(0, 3, nearest_c2)                # interp(c2) = 真实值，非 fill=8
tdSql.checkData(0, 4, True)
```

File: test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py [hunk 3]
```diff
@@ -6111,10 +6111,25 @@ def is_nearest(self, select_results, irowts_origin, irowts):
...
+                    tdLog.exit(f"_irowts should not be None when _irowts_origin is None: {row}, {sql}")
+                if row[-1] != True:
+                    self.check_failed = True
+                    tdLog.exit(f"_isfilled should be True when _irowts_origin is None: {row}, {sql}")
+                if fill_values is not None:
+                    for i, fv in enumerate(fill_values):
+                        if row[2 + i] != fv:
+                            self.check_failed = True
+                            tdLog.exit(f"interp col[{i}] should be {fv} but got {row[2 + i]} when _irowts_origin is None: {row}, {sql}")
+                continue
```
Comment (dapan1121) [inline test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py:6132]: **[MAJOR] `check_result_for_near` 对 `_irowts_origin is not None` 分支不校验 interp 列值**

NULL 分支（上面这段）在 `continue` 前正确验证了 fill 值。但 `continue` 之后的非 NULL 分支只调用 `is_nearest` 检查时间戳距离，**没有验证 `interp(c1)` / `interp(c2)` 是否等于最近邻数据行的实际列值**。

若 interp 函数错误地返回了 fill 默认值而非真实列值，这个检查完全不会发现。

建议在 `is_nearest` 通过后额外验证插值列与最近邻行的实际数据一致：
```python
if not self.is_nearest(select_all_results, irowts_origin, irowts):
    self.check_failed = True
    tdLog.exit(f"interp result is not the nearest for row: {row}, {sql}")

# 新增：验证 interp 列值等于最近邻行的实际数据
idx = self.binary_search_ts(select_all_results, irowts_origin)
for col_offset in range(len(row) - 3):  # col 2..N-2 是 interp 列
    expected = select_all_results[idx][1 + col_offset]
    actual = row[2 + col_offset]
    if actual != expected:
        self.check_failed = True
        tdLog.exit(f"interp col[{col_offset}] expected {expected} but got {actual}, row: {row}, {sql}")
```

File: test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py [hunk 4]
```diff
@@ -6386,6 +6407,47 @@ def check_interp_fill_extension(self):
         tdSql.checkData(0, 1, last_ts)
         tdSql.checkData(0, 4, False)
 
+        ### WHERE + range(point, duration): _irowts_origin should be NULL when nearest data exceeds surroundingTime
+        ### data in meters: first(ts) ~ 2018-09-17 09:00:00, last(ts) ~ 2018-09-17 10:23:19
```
Comment (dapan1121) [inline test/cases/11-Functions/04-Timeseries/test_fun_ts_interp.py:6434]: **[MAJOR] `~`（约等于）暗示用例边界存在不确定性，可能在数据分布不符合预期时偶发失败**

注释说 `first(ts) ~ 2018-09-17 09:00:00`，实际精确值是 `09:00:00.047`（由后面的动态查询获得）。第4个新增用例（非 NULL 分支）中，WHERE 下界 `'2018-09-17 10:00:00'` 是硬编码的——若 `[10:00, 10:23]` 内的最近数据点距离插值点 09:30 的距离实际超过 1h，该用例就会返回 NULL 而非非 NULL，与注释预期相反，且 `tdSql.checkRows(1)` 仍会通过而没有明确报错。

建议：
1. 通过查询确认 `[10:00, 10:23]` 内确实存在数据，并断言最近点与 09:30 的时间差确实 < 1h
2. 或动态计算插值点和 surroundingTime，使边界约束在任何测试数据版本下均成立

**Review by guanshengliang — APPROVED** [review]
